File: 1#._A._LATIHAN_KHUSUS/2._LATIHAN_DULU/Percabangan_Perulangan_Fungsi_dan_Error_dengan_fungsi.py
# Switch Case 1

def eru (l):

     match (l):

          case 1:
               print ("Angka 1")

          case 2:
               print ("Angka 2")

          case 3:
               print ("Angka 3")

          case 4:
               print ("Angka 4")

          case _:
               print ("Angka lain")

eru (1)
eru (2)
eru (3)
eru (4)
eru (5)


print ("\n --- batas --- \n")



# Switch Case 2

def hun (e):

     match (e):

          case "Merah":
               print ("Warna Merah")

          case "Kuning":
               print ("Warna Kuning")

          case "Hijau":
               print ("Warna Hijau")

          case _:
               print ("Warna lain")

hun ("Merah")
hun ("Kuning")
hun ("Hijau")


print ("\n --- batas --- \n")





# Fungsi dengan Percabangan Dasar 

def run (a):

     if a >= 5:
          print (f"Besar, angka a = {a}")

     else:
          print (f"Kecil, angka a = {a}")

run (10)
run (9)
run (8)
run (7)
run (6)
run (5)
run (4)
run (3)
run (2)
run (1)


print ("\n --- batas --- \n")




# Fungsi dengan percabangan lanjutan 2 

def tur (b):

     if b >= 8:
          print (f"Besar, angka b = {b}")

     elif b >= 5:
          print (f"Tengah, angka b = {b}")

     else:
          print (f"Kecil, angka b = {b}")

tur (10)
tur (9)
tur (8)
tur (7)
tur (6)
tur (5)
tur (4)
tur (3)
tur (2)


print ("\n --- batas --- \n")



# Fungsi dengan percabangan nilai rapor

def rapor (f):

     if f >= 95:
          print (f"A, nilai = {f}")

     elif f >= 90:
          print (f"B, nilai = {f}")

     elif f >= 80:
          print (f"C, nilai = {f}")

     elif f >= 70:
          print (f"D, nilai = {f}")

     elif f >= 60:
          print (f"E, nilai = {f}")

     elif f >= 50:
          print (f"Jelek amat, nilai = {f}")

rapor (100)
rapor (90)
rapor (80)
rapor (70)
rapor (60)
rapor (50)
rapor (40)
rapor (30)
rapor (20)
rapor (10)
rapor (5)


print ("\n --- batas --- \n")



# Fungsi dengan Percabangan nested 1 

def wer (c):

     cek = True

     if c >= 5:
          if cek:
               print (f"Besar, angka c = {c}")

          else:
               print (f"Tengah, angka c = {c}")

     else:
          print (f"Kecil, angka c = {c}")

wer (10)
wer (9)
wer (8)
wer (7)
wer (6)
wer (5)
wer (4)
wer (3)
wer (2)
wer (1)


print ("\n --- batas --- \n")




# Fungsi dengan Percabangan Nested 2 

def kop (f):

     cek = True

     if f >= 8:
          if cek:
               print (f"Besar, angka f = {f}")

          elif f >= 5:
               print (f"Tengah, angka f = {f}")

     else:
          print (f"Kecil, angka f = {f}")

kop (10)
kop (9)
kop (8)
kop (7)
kop (6)
kop (5)
kop (4)
kop (3)
kop (2)
kop (1)


print ("\n --- batas --- \n")




# Usia produktif manusia 

def usia (m):

     if m >= 15 and m <= 40:
          print (f"Usia yang produktif, usia = {m}")

     elif m > 40:
          print (f"Usia sudah cukup, usia = {m}")

     else:
          print (f"Masih kecil usiannya, usia = {m}")

usia (10)
usia (40)
usia (30)
usia (20)
usia (70)
usia (75)
usia (4)


print ("\n --- batas --- \n")



# Usia masuk JKT48 

def oshi (j):

     if j >= 13 and j <= 19:
          print (f"Usia yang sudah boleh masuk jkt48, usia = {j}")

     elif j >= 19:
          print (f"Sudah lebih dari cukup, usia = {j}")

     else:
          print (f"Masih kecil usiannya, usia = {j}")

oshi (20)
oshi (19)
oshi (18)
oshi (17)
oshi (16)
oshi (15)
oshi (14)
oshi (13)
oshi (12)


print ("\n --- batas --- \n")




# Usia masuk kerja

def ban (k):

     if k >= 15 and k <= 40:
          print (f"usia yang boleh kerja, usia = {k}")

     elif k > 40:
          print (f"Sudah pensiun, usia = {k}")

     else:
          print (f"Masih kecil usiannya, usia = {k}")

ban (70)
ban (60)
ban (50)
ban (40)
ban (30)
ban (20)
ban (10)
ban (5)


print ("\n --- batas --- \n")



# For dasar 

for a in range (1, 11):
     print (f"Urutan ke - {a}")


print ("\n --- batas --- \n")



# For dasar 

for i in range (11):
     print (f"Urutan ke - {i}")


print ("\n --- batas --- \n")




# Struktur data

dat = ["Halo Dunia", "Halo Fire", "Halo World", "Fireball"]

for i in dat:
     print (i)


print ("\n --- batas --- \n")

File: 1#._A._LATIHAN_KHUSUS/2._LATIHAN_DULU/Percabangan_Perulangan_Fungsi_dan_Error_dengan_fungsi.js
// Switch Case 1

function dek (m) {

     switch (m) {

          case 1:
               console.log ("Angka 1")
               break

          case 2:
               console.log ("Angka 2")
               break

          case 3:
               console.log ("Angka 3")
               break

          case 4:
               console.log ("Angka 4")
               break

          default:
               console.log ("Angka lain")
     }
}

dek (1)
dek (2)
dek (3)
dek (4)
dek (5)


console.log ("\n --- batas --- \n")




// Switch Case 2

function ger (p) {

     switch (p) {

          case "Merah":
               console.log ("Warna merah")
               break

          case "Kuning":
               console.log ("Warna kuning")
               break

          case "Hijau":
               console.log ("Warna hijau")
               break

          default:
               console.log ("Warna lain")
     }
}

ger ("Merah")
ger ("Kuning")
ger ("Hijau")
ger ("Hitam")


console.log ("\n --- batas --- \n")




// Fungsi dengan percabangan dasar

function dasar (a) {

     if (a >= 5) {
          console.log (`Besar, angka a = ${a}`)
     }

     else {
          console.log (`Kecil, angka a = ${a}`)
     }
}

dasar (10)
dasar (9)
dasar (8)
dasar (7)
dasar (6)
dasar (5)
dasar (4)
dasar (3)
dasar (2)
dasar (1)


console.log ("\n --- batas --- \n")




// Fungsi dengan percabangan lanjutan 

function lan (e) {

     if (e >= 8) {
          console.log (`Besar, angka e = ${e}`)
     }

     else if (e >= 5) {
          console.log (`Tengah, angka e = ${e}`)
     }

     else {
          console.log (`Kecil, angka e = ${e}`)
     }
}

lan (10)
lan (9)
lan (8)
lan (7)
lan (6)
lan (5)
lan (4)
lan (3)
lan (2)
lan (1)


console.log ("\n --- batas --- \n")




// Fungsi dengan percabangan rapor 

function rapor (f) {

     if (f >= 95) {
          console.log (`A, nilai = ${f}`)
     }

     else if (f >= 90) {
          console.log (`B, nilai = ${f}`)
     }

     else if (f >= 80) {
          console.log (`C, nilai = ${f}`)
     }

     else if (f >= 70) {
          console.log (`D, nilai = ${f}`)
     }

     else if (f >= 60) {
          console.log (`E, nilai = ${f}`)
     }

     else if (f >= 50) {
          console.log (`F, nilai = ${f}`)
     }
}

rapor (100)
rapor (90)
rapor (80)
rapor (70)
rapor (60)
rapor (50)
rapor (40)
rapor (30)
rapor (20)
rapor (10)
rapor (5)


console.log ("\n --- batas --- \n")



// Fungsi dengan Percabangan Nested 1 

function der (d) {
     
     cek = true

     if (d >= 5) {
          if (cek) {
               console.log (`Besar, angka d = ${d}`)
          }

          else {
               console.log (`Tengah, angka d = ${d}`)
          }
     }

     else {
          console.log (`Kecil, angka e = ${e}`)
     }
}

der (10)
der (9)
der (8)
der (7)
der (6)
der (5)
der (4)
der (3)
der (2)
der (1)


console.log ("\n --- batas --- \n")




// Fungsi dengan Percabangan Nested 2 

function fer (k) {

     cek = true

     if (k >= 8) {
          if (cek) {
               console.log (`Besar, angka k = ${k}`)
          }

          else if (k >= 5) {
               console.log (`Tengah, angka k = ${k}`)
          }
     }

     else {
          console.log (`Kecil, angka k = ${k}`)
     }
}

fer (10)
fer (9)
fer (8)
fer (7)
fer (6)
fer (5)
fer (4)
fer (3)
fer (2)
fer (1)


console.log ("\n --- batas --- \n")




// Fungsi Usia Produktif manusia

function pro (j) {

     if (j >= 15 && j <= 40) {
          console.log (`Usia sudah produktif, usia = ${j}`)
     }

     else if (j > 40) {
          console.log (`Sudah tua usiannya, usia = ${j}`)
     }

     else {
          console.log (`Masih kecil usiannya, usia = ${j}`)
     }
}

pro (50)
pro (40)
pro (30)
pro (20)
pro (10)
pro (5)
pro (3)


console.log ("\n --- batas --- \n")




// Fungsi dengan Percabangan Usia masuk JKT48

function der (e) {

     if (e >= 13 && e <= 19) {
          console.log (`Usia yang sudah boleh, usia = ${e}`)
     }

     else if (e > 19) {
          console.log (`Sudah lebih dari cukup, usia = ${e}`)
     }

     else {
          console.log (`Masih kecil usiannya, usia = ${e}`)
     }
}

der (30)
der (20)
der (10)
der (5)
der (2)


console.log ("\n --- batas --- \n")




// For dasar 

for (a = 1; a < 11; a++) {
     console.log (`Urutan ke - ${a}`)
}


console.log ("\n --- batas --- \n")



// For dasar 2 

for (b = 0; b < 11; b++) {
     console.log (`Urutan ke - ${b}`)
}


console.log ("\n --- batas --- \n")




// Struktur data

var dat = ["Halo Dunia", "Halo Fire", "Halo World", "Fireball"]

for (a = 0; a < dat.length; a++) {
     console.log (dat [a])
}


console.log ("\n --- batas --- \n")




// Struktur data 2

var der = ["Valid", "Notch", "Fall", "None"]

for (b = 0; b < der.length; b++) {
     console.log (der [b])
}


console.log ("\n --- batas --- \n")